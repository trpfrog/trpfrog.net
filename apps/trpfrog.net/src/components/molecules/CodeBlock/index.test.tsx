import { render } from '@testing-library/react'

import { CodeBlock, CodeBlockProps } from '.'
import { describe, expect, test } from 'vitest'

describe('CodeBlock', () => {
  test('snapshot test', async () => {
    // HACK to test RSC using testing-library
    // https://github.com/testing-library/react-testing-library/issues/1209
    const props: CodeBlockProps = {
      language: 'ts',
      children: `
        function fizzBuzz(n: number) {
          if (n % 15 === 0) {
            return 'FizzBuzz'
          } else if (n % 3 === 0) {
            return 'Fizz'
          } else if (n % 5 === 0) {
            return 'Buzz'
          } else {
            return n.toString()
          }
        }
      `,
    }
    const Result = await CodeBlock(props)
    const view = render(Result)
    expect(view.asFragment()).toMatchSnapshot()
  })
})
