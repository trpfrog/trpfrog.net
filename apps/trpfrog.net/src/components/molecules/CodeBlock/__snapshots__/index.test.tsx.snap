// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CodeBlock > snapshot test 1`] = `
<DocumentFragment>
  <div>
    <div
      class="tw-flex tw-h-fit tw-w-full tw-items-center tw-justify-between tw-rounded-t-lg tw-bg-trpfrog-500 tw-px-4 tw-py-1 dark:tw-bg-trpfrog-600"
    >
      <span
        class="tw-text-xs tw-font-bold tw-text-white"
      >
        TypeScript
      </span>
      <div
        class="tw-flex tw-gap-2"
      >
        <span
          data-tooltip-id="«r0»"
        >
          <button
            class="tw-grid tw-place-items-center tw-text-white tw-opacity-50 hover:tw-opacity-100"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-clone "
              data-icon="clone"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
            >
              <path
                d="M288 448l-224 0 0-224 48 0 0-64-48 0c-35.3 0-64 28.7-64 64L0 448c0 35.3 28.7 64 64 64l224 0c35.3 0 64-28.7 64-64l0-48-64 0 0 48zm-64-96l224 0c35.3 0 64-28.7 64-64l0-224c0-35.3-28.7-64-64-64L224 0c-35.3 0-64 28.7-64 64l0 224c0 35.3 28.7 64 64 64z"
                fill="currentColor"
              />
            </svg>
          </button>
        </span>
      </div>
    </div>
    <div
      class="tw-w-full tw-overflow-clip tw-overflow-x-scroll tw-font-mono tw-text-[0.8em] sp:tw-text-xs tw-border tw-border-trpfrog-500 dark:tw-border-trpfrog-600 tw-rounded-b-lg"
    >
      <pre
        class="shiki sp:tw-py-3 tw-py-3 tw-w-full tw-min-w-max shiki-themes github-light github-dark"
        style="background-color:#fff;--shiki-dark-bg:#24292e;color:#24292e;--shiki-dark:#e1e4e8"
        tabindex="0"
      >
        <code>
          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          />
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
                      function
            </span>
            <span
              style="color:#6F42C1;--shiki-dark:#B392F0"
            >
               fizzBuzz
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
              (
            </span>
            <span
              style="color:#E36209;--shiki-dark:#FFAB70"
            >
              n
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
              :
            </span>
            <span
              style="color:#005CC5;--shiki-dark:#79B8FF"
            >
               number
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
              ) {
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
                        if
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
               (n 
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
              %
            </span>
            <span
              style="color:#005CC5;--shiki-dark:#79B8FF"
            >
               15
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
               ===
            </span>
            <span
              style="color:#005CC5;--shiki-dark:#79B8FF"
            >
               0
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
              ) {
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
                          return
            </span>
            <span
              style="color:#032F62;--shiki-dark:#9ECBFF"
            >
               'FizzBuzz'
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
                        } 
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
              else
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
               if
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
               (n 
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
              %
            </span>
            <span
              style="color:#005CC5;--shiki-dark:#79B8FF"
            >
               3
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
               ===
            </span>
            <span
              style="color:#005CC5;--shiki-dark:#79B8FF"
            >
               0
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
              ) {
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
                          return
            </span>
            <span
              style="color:#032F62;--shiki-dark:#9ECBFF"
            >
               'Fizz'
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
                        } 
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
              else
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
               if
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
               (n 
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
              %
            </span>
            <span
              style="color:#005CC5;--shiki-dark:#79B8FF"
            >
               5
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
               ===
            </span>
            <span
              style="color:#005CC5;--shiki-dark:#79B8FF"
            >
               0
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
              ) {
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
                          return
            </span>
            <span
              style="color:#032F62;--shiki-dark:#9ECBFF"
            >
               'Buzz'
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
                        } 
            </span>
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
              else
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
               {
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#D73A49;--shiki-dark:#F97583"
            >
                          return
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
               n.
            </span>
            <span
              style="color:#6F42C1;--shiki-dark:#B392F0"
            >
              toString
            </span>
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
              ()
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
                        }
            </span>
          </span>
          

          <span
            class="line tw-inline-block tw-w-full tw-px-4 sp:tw-px-3"
          >
            <span
              style="color:#24292E;--shiki-dark:#E1E4E8"
            >
                      }
            </span>
          </span>
        </code>
      </pre>
    </div>
  </div>
</DocumentFragment>
`;
